library/MedEx/API.php

```diff
@@ -0,0 +1,2092 @@
+                    // Yes, so do this only for recalls that are in the past!  T
+                   
+                    if (strtotime($recall['r_eventDate']) < mktime(0,0,0)) {
+                        if ($this->recursive_array_search("recall_".$recall['r_pid'],$appt3)) { continue; }
+                    }
+                     $count_recalls++;
+                    $recall2 = array();                    
+                    $recall2['pc_pid']        = $recall['r_pid'];
+                    $recall2['e_C_UID']       = $event['C_UID'];
+                    $recall2['pc_eventDate']  = $recall['r_eventDate'];
+                    $recall2['pc_startTime']  = '10:42:00';
+                    $recall2['pc_eid']        = "recall_".$recall['r_pid'];
+                    $recall2['e_pc_eid']      = "recall_".$recall['r_pid'];
+                    $recall2['pc_aid']        = $recall['r_provider'];
+                    $recall2['e_reason']      = (!empty($recall['r_reason']))?:'';
+                    $recall2['e_is_subEvent_of']= "0";
+                    $recall2['language']      = $recall['language'];
+                    $recall2['pc_facility']   = $recall['r_facility'];
+                    $recall2['fname']         = $recall['fname'];
+                    $recall2['lname']         = $recall['lname'];
+                    $recall2['mname']         = $recall['mname'];
+                    $recall2['street']        = $recall['street'];
+                    $recall2['postal_code']   = $recall['postal_code'];
+                    $recall2['city']          = $recall['city'];
+                    $recall2['state']         = $recall['state'];
+                    $recall2['country_code']  = $recall['country_code'];
+                    $recall2['phone_home']    = $recall['phone_home'];
+                    $recall2['phone_cell']    = $recall['phone_cell'];
+                    $recall2['email']         = $recall['email'];
+                    $recall2['C_UID']         = $event['C_UID'];
+                    $recall2['E_fire_time']   = $event['E_fire_time'];
+                    $recall2['time_order']    = $event['time_order'];
+                    $recall2['M_type']        = $event['M_type'];
+                    $recall2['reply']         = "To Send";
+                    $recall2['extra']         = "QUEUED";
+                    $recall2['status']        = "SENT";
+                    $recall2['to']            = $results;
+                    
+                    $appt3[] = $recall2;
+                }
+            }   
+        }
+        if (!empty($RECALLS_completed)) {
+            $hipaa = $this->process_deletes($token,$RECALLS_completed);
+        }
+        if (!empty($appt3)) {
+            $hipaa = $this->process($token,$appt3);
+        }     
+
+        $responses['count_appts'] = $count_appts;
+        $responses['count_recalls'] = $count_recalls;
+        return $responses;
```

Comment:
There is a preference page for MedEx within this file.  The user doesn't need to see it if they are not using medexbank.com - ie. if they are using the Recall Board without the email/sms/avm there are no preferences to set.  If they are using MedEx, the options they use are either set in the preference function (File->Preferences in the bootstrap menu, which is "public function preferences" in the API) or on MedExbank.com itself.  I tried to stay away from touching anything else in the codebase for 3 reasons.  One, some of it makes no sense to me.  Two,  I imagine people will strip out portions of this code and make their own messaging service, which will require them to use the Globals sections like sms gateway etc fields.  Three, I wanted this to be able to be truly optional - like a module - easily deployed and easily removed.  So the less it touches other sections the better.  The only exceptions I couldn't avoid were the flow board (which Terry has been great about helping me hack it up to pieces and insert stuff) and the Calendar.